Userland/Libraries/LibWeb/Streams/ReadableStream.h

```diff
@@ -26,6 +27,13 @@ struct ReadableStreamGetReaderOptions {
     Optional<Bindings::ReadableStreamReaderMode> mode;
 };
 
+struct StreamPipeOptions {
+    bool prevent_close { false };
+    bool prevent_abort { false };
+    bool prevent_cancel { false };
+    Optional<JS::NonnullGCPtr<DOM::AbortSignal>> signal;
```

Comment:
It's a bit curious to have an `Optional<Nonnull` option. Could this just be a GCPtr instead? But as I read the IDL- when is this optional :thonk:

Reply:
Huh, I just built locally, seems like just an IDL generator thing. Wonder if that's intentional, or just from the general rule to use Optional

Reply:
Also not sure, if this is an IDL generator thing or intentional. Didn't investigate further for this PR though. Would probably align better with the spec steps if this and similar could just be `JS:GCPtr<T>`.

Reply:
Yeah I think it is an IDL thing and we would unfortusntely need to change IDL generator to fix. So maybe easier to leave for later on, since I imagine we would have to change other places too

Reply:
Re-reading the code for `pipe_to` and its AO code seems specs steps actually align better with it being an `Optional<T>`. I should probably change the `Optional<JS::Value> signal` to `JS::Value signal` on AO `readable_stream_pipe_to()`

Reply:
Or not, it's getting to late for me now...

Reply:
i think it is good as is :^)

Reply:
The same pattern seems to be used in other places as well, such as: `Userland/Libraries/LibWeb/DOM/IDLEventListener.h`

Reply:
I made https://github.com/SerenityOS/serenity/pull/23877 for this